components/tidb_query_vec_executors/src/runner.rs

```diff
@@ -469,20 +434,148 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
             }
 
             // Grow batch size
-            if batch_size < BATCH_MAX_SIZE {
-                batch_size *= BATCH_GROW_FACTOR;
-                if batch_size > BATCH_MAX_SIZE {
-                    batch_size = BATCH_MAX_SIZE
-                }
-            }
+            grow_batch_size(&mut batch_size);
         }
     }
 
+    pub fn handle_streaming_request(
+        &mut self,
+    ) -> Result<(Option<(StreamResponse, IntervalRange)>, bool)> {
+        let mut warnings = self.config.new_eval_warnings();
+
+        let (mut record_len, mut is_drained) = (0, false);
+        let mut chunk = Chunk::default();
+        let mut ctx = EvalContext::new(self.config.clone());
+        let batch_size = self.stream_row_limit.min(BATCH_MAX_SIZE);
+
+        // record count less than batch size and is not drained
+        while record_len < self.stream_row_limit && !is_drained {
```

Comment:
Instead of `record_len`, I prefer using `processed_row_len` as the threshold. Considering a Top1 operator, the executor will process all the rows until the end because `record_len` is always 0.

Reply:
Or you can set a time threshold like non-stream `handle_request`

Reply:
> Instead of `record_len`, I prefer using `processed_row_len` as the threshold. Considering a Top1 operator, the executor will process all the rows until the end because `record_len` is always 0.

I think in you scenario, process all rows should be the expected behavior. @iosmanthus PTAL

Reply:
Found that `BatchExecutorsRunner` ignores 
https://github.com/tikv/tikv/blob/a0600af2685cb20a21cffe37276b1a7a60609521/src/server/config.rs#L89
so I think
https://github.com/tikv/tikv/blob/a0600af2685cb20a21cffe37276b1a7a60609521/src/server/config.rs#L90
can also be ignored.

Then both `handle_request` and `handle_streaming_request` can use time slice as their runtime limit, which I think it's more unified.

Reply:
> I think in you scenario, process all rows should be the expected behavior.

We cannot block the thread pool. And it's not stream-ish to finish a request at once.